src/main/java/it/pagopa/transactions/utils/TransactionsUtils.java

```diff
@@ -10,14 +10,42 @@
 import org.springframework.stereotype.Component;
 import reactor.core.publisher.Mono;
 
+import java.util.EnumMap;
+import java.util.Map;
+
 @Component
 public class TransactionsUtils {
 
     private final TransactionsEventStoreRepository<Object> eventStoreRepository;
 
+    private final Map<it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto, it.pagopa.generated.transactions.server.model.TransactionStatusDto> transactionStatusLookupMap = new EnumMap<>(
+            it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto.class
+    );
+
     @Autowired
     public TransactionsUtils(TransactionsEventStoreRepository<Object> eventStoreRepository) {
         this.eventStoreRepository = eventStoreRepository;
+        init();
+    }
+
+    private void init() {
+        for (it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto enumValue : it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto
+                .values()) {
+            /*
+             * @formatter:off
+             * This lookup map serve two purpose:
+             * 1) handle enumerations conversion from commons and transactions-service TransactionStatusDto enumerations
+             * 2) in case one enumeration from commons is not present into transactions-service's ones
+             * then an IllegalArgumentException is raised, preventing the module from being
+             * run and avoiding runtime errors correlated to specs updated into commons
+             * that are not reflected into transactions specs (such as a transaction status added only into commons)
+             * @formatter:on
+             */
+            transactionStatusLookupMap.put(
+                    enumValue,
+                    it.pagopa.generated.transactions.server.model.TransactionStatusDto.fromValue(enumValue.toString())
+            );
+        }
     }
```

Comment:
Does it need to be a separate method?
If so, I would rename it to a more explicit name, like `initializeStatusMap` or something like that